transaction-context/src/lib.rs

```diff
@@ -248,14 +272,31 @@ impl<'ix_data> TransactionContext<'ix_data> {
         instruction_data: Cow<'ix_data, [u8]>,
     ) -> Result<(), InstructionError> {
         debug_assert_eq!(deduplication_map.len(), MAX_ACCOUNTS_PER_TRANSACTION);
+        let trace_len = self.instruction_trace.len();
+        let instruction_index = trace_len.saturating_sub(1);
+        let penultimate_instruction_accounts = self
+            .instruction_trace
+            .get(trace_len.saturating_sub(2))
+            .map(|item| item.instruction_accounts);
+
         let instruction = self
             .instruction_trace
             .last_mut()
             .ok_or(InstructionError::CallDepth)?;
+
         instruction.program_account_index_in_tx = program_index;
-        instruction.instruction_accounts = instruction_accounts;
-        instruction.instruction_data = instruction_data;
-        instruction.dedup_map = deduplication_map;
+        instruction.configure_vm_slices(
+            instruction_index as u64,
+            trace_len as u64,
+            penultimate_instruction_accounts,
+            instruction_accounts.len(),
+            instruction_data.len() as u64,
+        );
+        self.instruction_accounts
+            .extend_from_slice(&instruction_accounts);
+        self.deduplication_maps
+            .extend_from_slice(&deduplication_map);
```

Comment:
These memory copies here annoy me, but I already have a plan to remove them. I'll first get this PR merged and then address that in a follow-up.